core/src/main/java/org/apache/druid/segment/column/ColumnType.java

```diff
@@ -26,18 +26,71 @@
 
 import javax.annotation.Nullable;
 
+/**
+ * Native Druid types.
+ *
+ * @see TypeSignature
+ */
 @JsonSerialize(using = ToStringSerializer.class)
 public class ColumnType extends BaseTypeSignature<ValueType>
 {
+  /**
+   * Druid string type. Values will be represented as {@link String} or {@link java.util.List<String>} in the case
+   * of multi-value string columns. {@link ColumnType} has insufficient information to distinguish between single
+   * and multi-value strings, this requires a type 'ColumnCapabilities' which is available at a higher layer.
```

Comment:
Link to the specific hasMultipleValues function? Or mention, if it's not available from this package.

Reply:
not in this package unfortunately but will do. i do still someday wish to merge druid-core into druid-processing since i find the split a bit frustrating, but that is so disruptive i should probably give up hope that it will ever happen :trollface:

Reply:
> so disruptive i should probably give up hope that it will ever happen

There's always the old "do it during a quiet week" strategy 🙂